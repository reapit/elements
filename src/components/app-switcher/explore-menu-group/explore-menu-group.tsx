import React from 'react'
import { Menu } from '../../menu'
import { AppMenuGroupHasAccessContext } from '../menu-group-context'

type AppSwitcherMenuGroupProps = {
  children: React.ReactNode
}

function AppSwitcherExploreMenuGroup({ children }: AppSwitcherMenuGroupProps) {
  return (
    <Menu.Group label={'EXPLORE'}>
      <AppMenuGroupHasAccessContext.Provider value={false}>{children}</AppMenuGroupHasAccessContext.Provider>
    </Menu.Group>
  )
}

export { AppSwitcherExploreMenuGroup }
