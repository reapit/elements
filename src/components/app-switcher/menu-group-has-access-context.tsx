import { createContext, useContext } from 'react'

export const AppSwitcherMenuGroupHasAccessContext = createContext<boolean>(false)

export function useAppSwitcherMenuGroupHasAccessContext() {
  return useContext(AppSwitcherMenuGroupHasAccessContext)
}
