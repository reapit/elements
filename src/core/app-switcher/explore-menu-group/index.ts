export * from './explore-menu-group'
