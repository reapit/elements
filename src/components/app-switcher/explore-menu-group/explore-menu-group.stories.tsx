import type { Decorator, Meta, StoryObj } from '@storybook/react'
import { AppSwitcher } from '../app-switcher'
import { AppSwitcherExploreMenuGroup } from './explore-menu-group'

const useParentDecorator: Decorator = (Story) => {
  return (
    <div style={{ width: 'fit-content' }}>
      <Story />
    </div>
  )
}

const meta = {
  title: 'Components/AppSwitcher/AppSwitcherExploreAppsMenuGroup',
  component: AppSwitcherExploreMenuGroup,
  decorators: [useParentDecorator],
} satisfies Meta<typeof AppSwitcherExploreMenuGroup>

export default meta

type Story = StoryObj<typeof meta>

export const Default: Story = {
  args: {
    children: null,
  },
  render: ({}) => {
    return (
      <AppSwitcher.ExploreMenuGroup>
        <AppSwitcher.ReapitPMMenuItem url={'#'} />
        <AppSwitcher.ReapitSalesMenuItem url={'#'} />
        <AppSwitcher.KeyWhereMenuItem url={'#'} />
      </AppSwitcher.ExploreMenuGroup>
    )
  },
}

export const WhenAMenuItemIsFocused: Story = {
  args: {
    children: null,
  },
  render: ({}) => {
    return (
      <AppSwitcher.ExploreMenuGroup>
        <AppSwitcher.ReapitPMMenuItem url={'#'} />
        <AppSwitcher.ReapitSalesMenuItem url={'#'} />
        <AppSwitcher.KeyWhereMenuItem url={'#'} />
      </AppSwitcher.ExploreMenuGroup>
    )
  },
}
