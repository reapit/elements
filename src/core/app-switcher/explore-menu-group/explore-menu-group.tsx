import { AppSwitcherMenuGroupHasAccessContext } from '../menu-group-has-access-context'
import { Menu } from '#src/deprecated/menu'

import type { ReactNode } from 'react'

interface AppSwitcherMenuGroupProps {
  children: ReactNode
}

function AppSwitcherExploreMenuGroup({ children }: AppSwitcherMenuGroupProps) {
  return (
    <Menu.Group label={'EXPLORE'}>
      <AppSwitcherMenuGroupHasAccessContext.Provider value={false}>
        {children}
      </AppSwitcherMenuGroupHasAccessContext.Provider>
    </Menu.Group>
  )
}

export { AppSwitcherExploreMenuGroup }
