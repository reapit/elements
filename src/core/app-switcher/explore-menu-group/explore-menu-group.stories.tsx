import type { Decorator, Meta, StoryObj } from '@storybook/react-vite'
import { AppSwitcher } from '../app-switcher'
import { AppSwitcherExploreMenuGroup } from './explore-menu-group'

const useParentDecorator: Decorator = (Story) => {
  return (
    <div style={{ boxSizing: 'content-box', border: '1px solid #FA00FF' }}>
      <Story />
    </div>
  )
}

const meta = {
  title: 'Core/AppSwitcher/ExploreMenuGroup',
  component: AppSwitcherExploreMenuGroup,
  argTypes: {
    children: {
      control: false,
    },
  },
  decorators: [useParentDecorator],
} satisfies Meta<typeof AppSwitcherExploreMenuGroup>

export default meta

type Story = StoryObj<typeof meta>

export const Default: Story = {
  args: {
    children: null,
  },
  render: ({}) => {
    return (
      <AppSwitcher.ExploreMenuGroup>
        <AppSwitcher.ProductMenuItem href={globalThis.top?.location.href!} productId="ireWeb" />
      </AppSwitcher.ExploreMenuGroup>
    )
  },
}
