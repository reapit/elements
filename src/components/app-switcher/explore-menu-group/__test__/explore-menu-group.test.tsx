import { render } from '@testing-library/react'
import { AppSwitcherExploreMenuGroup } from '../explore-menu-group'
import { AppSwitcherMenuGroupHasAccessContext } from '../../menu-group-has-access-context'

test('renders AppSwitcherExploreMenuGroup properly', () => {
  const { asFragment } = render(<AppSwitcherExploreMenuGroup>Fake child</AppSwitcherExploreMenuGroup>)
  expect(asFragment()).toMatchSnapshot()
})

test('provides an `AppSwitcherMenuGroupHasAccessContext` value of `false`', () => {
  let hasAccess: boolean | undefined

  render(
    <AppSwitcherExploreMenuGroup>
      <AppSwitcherMenuGroupHasAccessContext.Consumer>
        {(value) => {
          hasAccess = value
          return null
        }}
      </AppSwitcherMenuGroupHasAccessContext.Consumer>
    </AppSwitcherExploreMenuGroup>,
  )

  expect(hasAccess).toBe(false)
})
