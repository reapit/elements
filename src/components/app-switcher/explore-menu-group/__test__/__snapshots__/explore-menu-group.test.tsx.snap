// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders AppSwitcherExploreMenuGroup properly 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-11 el-menu-item-group"
    role="group"
  >
    <div
      class="mocked-styled-10 el-menu-item-group-title"
    >
      EXPLORE
    </div>
    Fake child
  </div>
</DocumentFragment>
`;
